server/controllers/books.js

```diff
@@ -0,0 +1,217 @@
+            return res.status(409).send({ errors });
+          }
+          return res.status(400).send({ errors });
+        }
+
+        return res.status(500).send({
+          error
+        });
+      }
+      );
+  },
+
+  /**
+   * Method to get all books from the library
+   * @param {Object} req
+   * @param {Object} res
+   *
+   * @returns {Promise.<Object>} books
+   */
+  getAllBooks(req, res) {
+    if (req.auth.admin) {
+      attributes = [...attributes,
+        'isBorrowed',
+        'createdAt'
+      ];
+    }
+
+    return db.Book
+      .scope('active')
+      .findAll({
+        attributes,
+        include: [{
+          model: db.Category,
+          attributes: ['name', 'slug']
+        }],
+      })
+      .then(books => res.status(200).send({ books }))
+      .catch((error) => {
+        if (error) {
+          return res.status(500).send({
+            error: 'Request could not be processed, please try again later'
+          });
+        }
+      });
+  },
+
+  /**
+   * [getBook description]
+   * @method getBook
+   * @param  {[type]} req [description]
+   * @param  {[type]} res [description]
+   * @return {[type]}     [description]
+   */
+  getBook(req, res) {
+    if (req.auth.admin) {
+      attributes = [...attributes,
+        'isBorrowed',
+        'createdAt'
+      ];
+    }
+
+    if (!req.params.bookId || !Number.isInteger(parseInt(req.params.bookId, 10))) {
+      return res.status(400).send({ error: 'Book Id is invalid' });
+    }
+
+    return db.Book
+      .findOne({
+        attributes,
+        include: [{
+          model: db.Category,
+          attributes: ['name', 'slug'],
+        }],
+        where: { id: req.params.bookId }
+      })
+      .then((book) => {
+        if (!book) {
+          return res.status(404).send({ error: 'Book not found' });
+        }
+
+        return res.status(200).send({ book });
+      })
+      .catch((error) => {
+        if (error) {
+          return res.status(500).send({
+            error: 'Request could not be processed, please try again later'
+          });
+        }
+      });
+  },
+
+  /**
+   * This updates the book in the library
+   * @param {Object} req
+   * @param {Object} res
+   *
+   * @returns {Object} book
+   */
+  update(req, res) {
+    return db.Book
+      .findById(req.params.bookId)
+      .then((book) => {
+        if (!book) {
+          return res.status(404).send({ error: 'Book not found' });
+        }
+        book.update({
+          title: req.body.title,
+          categoryId: req.body.categoryId,
+          author: req.body.author,
+          description: req.body.description,
+          coverPic: req.body.coverPic,
+        }).then((result) => {
+          if (result) {
+            return res.status(200).send({
+              book: formatBookObject(book, req.category)
+            });
+          }
+        })
+          .catch((error) => {
+            if (error.name === 'SequelizeValidationError' ||
```

Comment:
The same issue with multiple code repetition., I can see the use of this code in multiple places